mono/btls/CMakeLists.txt

```diff
@@ -80,12 +80,8 @@ set (
 	btls-x509-verify-param.h
 	btls-x509.c
 	btls-x509.h
-
-	${BORINGSSL_OBJECTS}
 )
 
-if (BUILD_SHARED_LIBS)
-	add_library (mono-btls-shared SHARED ${MONO_BTLS_SOURCES})
-else ()
-	add_library (mono-btls-static STATIC ${MONO_BTLS_SOURCES})
-endif ()
+add_library (mono-btls-objects OBJECT ${MONO_BTLS_SOURCES})
+add_library (mono-btls-shared SHARED $<TARGET_OBJECTS:mono-btls-objects> ${BORINGSSL_OBJECTS})
```

Comment:
I assume the STATIC/SHARED flags to add_library added it. But since we are passing already built objects files to add_library now, we can't pass the flag. Have to check the build output (make VERBOSE=1) too see what is happening.